ush/setup.py

```diff
@@ -1640,284 +1179,97 @@ def get_location(xcs, fmt):
-        "N_VAR_LNDP": N_VAR_LNDP,
-        "LNDP_TYPE": LNDP_TYPE,
-        "LNDP_MODEL_TYPE": LNDP_MODEL_TYPE,
-        "FHCYC_LSM_SPP_OR_NOT": FHCYC_LSM_SPP_OR_NOT,
-    }
-
-    # write derived settings
-    cfg_d["derived"] = settings
+    if use_thompson:
+        fixed_files["CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING"].extend(
+            mapping
+        )
+        fixed_files["FIXgsm_FILES_TO_COPY_TO_FIXam"].extend(fix_files)
 
+        log_info(
+            f"""
+            Since the Thompson microphysics parameterization is being used by this
+            physics suite (CCPP_PHYS_SUITE), the names of the fixed files needed by
+            this scheme have been appended to the array FIXgsm_FILES_TO_COPY_TO_FIXam,
+            and the mappings between these files and the symlinks that need to be
+            created in the cycle directories have been appended to the array
+            CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING.  After these modifications, the
+            values of these parameters are as follows:
+
+            CCPP_PHYS_SUITE = \"{workflow_config["CCPP_PHYS_SUITE"]}\"
+            """
+        )
+        log_info(
+            f"""
+                FIXgsm_FILES_TO_COPY_TO_FIXam =
+                {list_to_str(fixed_files['FIXgsm_FILES_TO_COPY_TO_FIXam'])}
+            """
+        )
+        log_info(
+            f"""
+                CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING =
+                {list_to_str(fixed_files['CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING'])}
+            """
+        )
     #
     # -----------------------------------------------------------------------
     #
-    # NCO specific settings
-    #
-    # -----------------------------------------------------------------------
-    #
-    settings = {
-        "COMIN_BASEDIR": COMIN_BASEDIR,
-        "COMOUT_BASEDIR": COMOUT_BASEDIR,
-        "OPSROOT": OPSROOT,
-        "COMROOT": COMROOT,
-        "PACKAGEROOT": PACKAGEROOT,
-        "DATAROOT": DATAROOT,
-        "DCOMROOT": DCOMROOT,
-        "DBNROOT": DBNROOT,
-        "EXTROOT": EXTROOT,
-        "SENDECF": SENDECF,
-        "SENDDBN": SENDDBN,
-        "SENDDBN_NTC": SENDDBN_NTC,
-        "SENDCOM": SENDCOM,
-        "SENDWEB": SENDWEB,
-        "KEEPDATA": KEEPDATA,
-        "MAILTO": MAILTO,
-        "MAILCC": MAILCC,
-    }
-
-    cfg_d["nco"].update(settings)
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Now write everything to var_defns.sh file
+    # Generate var_defns.sh file in the EXPTDIR. This file contains all
+    # the user-specified settings from expt_config.
     #
     # -----------------------------------------------------------------------
     #
 
-    # update dictionary with globals() values
-    update_dict(globals(), cfg_d)
+
+    extend_yaml(expt_config)
+    for sect, sect_keys in expt_config.items():
+        for k, v in sect_keys.items():
+            expt_config[sect][k] = str_to_list(v)
+    extend_yaml(expt_config)
```

Comment:
After all the validation and value-updating, iterate through filling any remaining jinja templates that may exist in the configuration.